app/lib/redis_helper.rb

```diff
@@ -19,6 +19,7 @@ def self.decrement_or_initialize(key, &block)
 
   # The Hydration block needs to return an array of hashes that have an 'id' field
   def self.cache_info_by_ids(key_prefix, ids, &hydration_block)
+    return [] if ids.empty?
```

Comment:
This is getting close to what I want, but why is this explicit check even necessary in the first place?

If the user happens to pass in an empty array of ids, then the `keys` variable below maps over an empty array, which isn't forbidden. It simply returns an empty array.

That empty array is passed to `Rails.cache.read_multi`, can it deal with empty arrays? Ideally, I would just expect that method to return an empty hash into `info` as well. Then, `uncached_ids` will be empty, the `unless` block never gets entered, and `info.values` happily returns an empty array.

What am I missing here?

Reply:
It didn't work for some reason 2 days ago, but now I just removed it and it just works, so not sure what happened